specifyweb/specify/record_merging.py

```diff
@@ -308,7 +308,7 @@ def update_record(record: models.Model):
 
                 response: http.HttpResponse = update_record(obj)
                 if response is not None and response.status_code != 204:
```

Comment:
@CarolineDenis thanks for fixing this!

@acwhite211 on second thought, do we even need this now? I added handling the invalid http responses through FailedMergingException to preserve atomicity, which gets caught by the caller of this function. See [here](https://github.com/specify/specify7/blob/075104652b2204e273501289a7090e79b3998007/specifyweb/specify/record_merging.py#L45-L64).


So, we can get rid of all the HTTP responses [here]. (https://github.com/specify/specify7/blob/075104652b2204e273501289a7090e79b3998007/specifyweb/specify/record_merging.py#L150-L350). So now, this function won't return anything but would raise a FailedMergingException or any other exception. 

Here, now we define the 204 http in the try block, and return that.

https://github.com/specify/specify7/blob/075104652b2204e273501289a7090e79b3998007/specifyweb/specify/record_merging.py#L45-L64